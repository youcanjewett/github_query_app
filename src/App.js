import React, {useState} from "react";
import SearchBar from './components/searchBar';
import SearchResults from './components/searchResults';
import './style.css';

function App() {
  const [repos, updateRepos] = useState([]);

  // sorts the repos by star rating, or returns message if there are no repos
  let sortedRepos = repos.length !== 0 ? 
    repos.data.sort((a,b) => (b.stargazers_count > a.stargazers_count) ? 1 : -1) : 
    ['no repos here'];

  return (
    <div class="app_container">
      <h1 class = "app_title">GitHub Repo Search</h1>
      <div>
       <SearchBar updateRepos = {updateRepos}></SearchBar>
      </div>
      {sortedRepos.map((repo) => {
        return <SearchResults repo={repo} />
      })}
    </div>
  );
}

export default App;
