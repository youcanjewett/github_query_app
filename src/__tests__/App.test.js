import {render} from '@testing-library/react'
import renderer from 'react-test-renderer';
import App from './../App';

// snapshot test
describe("Jest Snapshot testing suite", () => {
    it("Matches Snapshot", () => {
      const page = renderer.create(<App />).toJSON();
      expect(page).toMatchSnapshot();
    });
  });

// unit test
it('renders the page', () => {
    render(<App />);
  });

  // you should also have additional tests to check that each component is 
  // rendering correctly, passing props correctly, state is updating as expected, 
  // etc.