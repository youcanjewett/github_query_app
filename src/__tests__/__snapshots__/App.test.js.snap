// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches Snapshot 1`] = `
<div
  class="app_container"
>
  <h1
    class="app_title"
  >
    GitHub Repo Search
  </h1>
  <div>
    <div>
      <input
        class="searchbar"
        onChange={[Function]}
        placeholder="Enter a GitHub username"
        required={true}
        size="30"
        type="text"
        value=""
      />
      <button
        class="submit_button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
  <div
    class="repo_items"
  >
    No repos yet
  </div>
</div>
`;
